src/main/java/seedu/address/logic/commands/DeleteCommissionCommand.java

```diff
@@ -0,0 +1,64 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_CUSTOMERS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.commission.Commission;
+
+/**
+ * Deletes the commission of an existing customer in the ArtBuddy.
+ */
+public class DeleteCommissionCommand extends Command {
+
+    public static final String COMMAND_WORD = "delcom";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the commission identified by the index number used in the displayed commission list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_COMMISSION_SUCCESS = "Deleted Commission: %1$s";
+
+    private final Index targetIndex;
+
+    public DeleteCommissionCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.hasActiveCustomer()) {
+            throw new CommandException(Messages.MESSAGE_NO_ACTIVE_CUSTOMER);
+        }
+
+        List<Commission> lastShownList = model.getFilteredCommissionList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_COMMISSION_DISPLAYED_INDEX);
+        }
+
+        Commission commissionToDelete = lastShownList.get(targetIndex.getZeroBased());
+        model.deleteCommission(commissionToDelete);
+
+        // This clones the person (shallow copy) so the GUI updates properly.
```

Comment:
Just a hypothesis: not sure if updating the hash and equality methods to include the commissions for `Customer` and avoiding placing the ObservableList within Customer would fix this issue with the GUI not being updated.

Reply:
Updated hash & equality methods but problem still persists - thus getClone() is kept for now.

Reply:
I took another look at this, I suspect the problem lies in the `AddressBook` `removeCommission` method modifying the original Customer instance, so the UI thinks the Customer has not been updated.

```java
// Current
    /**
     * Removes {@code key} from this {@code AddressBook}.
     * {@code key} must exist in the customer's commission list.
     */
    public void removeCommission(Customer customer, Commission key) {
        requireAllNonNull(customer, key);
        assert hasCommission(customer, key);
        getCustomerCommissions(customer).remove(key);
        commissions.remove(key);
    }
// Proposed
    /**
     * Removes {@code key} from this {@code AddressBook}.
     * {@code key} must exist in the customer's commission list.
     */
    public void removeCommission(Customer customer, Commission key) {
        requireAllNonNull(customer, key);
        assert hasCommission(customer, key);
        // make a copy of customer's commissions with key removed.
        Set<Commission> filteredCommissions = getCustomerCommissions(customer).stream()
                .filter(commission -> !commission.equals(key)).collect(Collectors.toSet());
        commissions.remove(key);
        // make a copy of customer with the updated commissions.
        Customer newCustomer = customer.copyWithCommissions(filteredCommissions);
        // replace original customer with updated customer to trigger customer list UI updates
        setCustomer(customer, newCustomer);
    }
```